elide-async/src/main/java/com/yahoo/elide/async/service/DefaultResultStorageEngine.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.async.service;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQueryResultStorage;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.request.EntityProjection;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Iterator;
+
+import javax.inject.Singleton;
+import javax.sql.rowset.serial.SerialBlob;
+
+/**
+ * Utility class which implements ResultStorageEngine.
+ */
+@Singleton
+@Slf4j
+@Getter
+public class DefaultResultStorageEngine implements ResultStorageEngine {
+
+    @Setter private Elide elide;
+    @Setter private DataStore dataStore;
+    private String baseURL;
+
+    public DefaultResultStorageEngine() {
+    }
+
+    public DefaultResultStorageEngine(Elide elide, DataStore dataStore, String baseURL) {
+        this.elide = elide;
+        this.dataStore = dataStore;
+        this.baseURL = baseURL;
+    }
+
+    @Override
+    public URL storeResults(String asyncQueryID, String responseBody) {
```

Comment:
We should change the method to take asyncQueryID and Byte[] as the input args. ProcessQuery should send byte[] instead of responseBody